feat(dialog): use native HTML dialog element (#2078)
RedHat-UX/red-hat-design-system: elements/rh-dialog/docs/40-accessibility.md

## Focus order

> When a dialog opens, the close dialog button recieves focus by default. Users have the ability to tab through each focusable element in the dialog. When the last focusable element in the dialog is reached, focus returns to the close dialog button—effectively trapping focus inside the dialog.

Reviewer: ```suggestion
When a dialog opens, the close dialog button receives focus by default. Users have the ability to tab through each focusable element in the dialog. When the last focusable element in the dialog is reached, focus returns to the close dialog button—effectively trapping focus inside the dialog.
```

### Backdrop

A dialog will close by users clicking or tapping the backdrop or outside of the container.

## Accessible labels

> Each dialog needs an accessible name. If a dialog has a heading tag in the `header` or default slot, this component will automatically apply an appropriate ID to the heading tag and an `aria-labelledby` attribute to the `<dialog>`.

Reviewer: ```suggestion
Each dialog needs an accessible name. If a dialog has a heading tag in the `header` or default slot, it will automatically be used as the accessible name of the dialog.
```

> Users can optionally provide an `accessible-label` attribute which overrides the built-in `aria-labelledby` functionality:

Reviewer: ```suggestion
If there is no slotted heading, users should provide an `accessible-label` attribute, the value of which will be used as the accessible name:
```

```html
<rh-dialog accessible-label="Page Properties" trigger="first-modal-trigger">
  ...
</rh-dialog>
```

> The `accessible-label` attribute assigns an `aria-label` to the `<dialog>` element.

Reviewer: ```suggestion
```

i don't think this information (the "how") is relevant to consumers. Rather, the relevant part is that applying `accessible-label` to rh-dialog makes the dialog more accessible to sr users.

same for the comment about `aria-labelledby` above. 

we are in the business of making accessible html elements with custom, public apis. the exact, granular details of what happens in the shadow root and in memory (ElementInternals) should not be of any concern to consumers of the design system.

Author: Resolved in [303f85f](https://github.com/RedHat-UX/red-hat-design-system/pull/2078/commits/303f85f107a053833d07585d07cff8bdd74f2cff).

> If neither an `accessible-label` nor any headings exist, the `aria-label` on the `<dialog>` will default to the text of the dialog's trigger.

Reviewer: ```suggestion
If neither an `accessible-label` nor any headings exist, the accessible name of the dialog will fall back to the text of the dialog's trigger.
```
